Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -96,6 +96,35 @@ final class CacheAdvanceTests: XCTestCase {
         XCTAssertEqual(messages, [])
     }
 
+  func test_messages_throwsFileCorruptedWhenOffsetInFileAtEndOfNewsetMessageOutOfSync() throws {
+      let randomHighValue: UInt64 = 10_1000
+      let header = try CacheHeaderHandle(
+        forReadingFrom: testFileLocation,
+        maximumBytes: randomHighValue,
+        overwritesOldMessages: true)
+      let cache = CacheAdvance<TestableMessage>(
+        fileURL: testFileLocation,
+        writer: try FileHandle(forWritingTo: testFileLocation),
+        reader: try CacheReader(forReadingFrom: testFileLocation),
+        header: try CacheHeaderHandle(
+            forReadingFrom: testFileLocation,
+            maximumBytes: header.maximumBytes,
+            overwritesOldMessages: header.overwritesOldMessages),
+        decoder: JSONDecoder(),
+        encoder: JSONEncoder())
+
+      // Make sure the header data is persisted before we read it as part of the `messages()` call below.
+      try header.synchronizeHeaderData()
+      // Our file is empty. Make the file corrupted by setting the offset at end of newest message to be further in the file.
+      // This should never happen, but past versions of this repo could lead to a file having this kind of inconsistency if a crash occurred at the wrong time.
+      try header.updateOffsetInFileAtEndOfNewestMessage(
+        to: FileHeader.expectedEndOfHeaderInFile + UInt64(MessageSpan.storageLength) + 1)
```

Comment:
If this file is empty would `FileHeader.expectedEndOfHeaderInFile + 1` cause the issue? I am trying to understand why we need to add `UInt64(MessageSpan.storageLength)`.

Reply:
`FileHeader.expectedEndOfHeaderInFile + 1` does indeed cause the issue, though I don't think that's what the file would have looked like when we were in the bad state. My guess is that we had at least one span in the expected end of newest message, plus the length of that message.

I'm happy to change this to just be `+ 1` since that does reproduce the issue, and since I think we've solve the original issue. I'll do that in a follow-up PR.